sdk/src/test/scala/com/horizen/account/performance/AccountSidechainNodeViewHolderPerfTest.scala

```diff
@@ -113,6 +113,163 @@ class AccountSidechainNodeViewHolderPerfTest
+      println(s"Total time $totalTime ms")
+      timePerTx = totalTime.toFloat / numOfTxs
+      println(s"Time per transactions ${timePerTx} ms")
+      println(
+        s"Average time per transactions in Snapshots ${listOfSnapshots.map(res => res.toFloat / numOfTxsPerSnapshot)} "
+      )
+      out.write(s"\n********************* Reverse order test results *********************\n")
+      out.write(s"Duration of the test:                      $totalTime ms\n")
+      out.write(s"Average time per transaction:             ${timePerTx} ms\n")
+      out.write(s"Average time per transaction in snapshots:\n")
+      listOfSnapshots.map(res => res.toFloat / numOfTxsPerSnapshot).zipWithIndex.foreach { case (res, idx) =>
+        out.write(s"Snapshot $idx: $res ms\n")
+      }
+      out.write(s"Number of transactions per snapshot: $numOfTxsPerSnapshot\n")
+
+      println("Starting test random order")
+      // Resetting MemPool
+      mempool = AccountMemoryPool.createEmptyMempool(() => state, () => state, mempoolSettings)
+      val randomOrderList = Random.shuffle(listOfTxs)
+      listOfSnapshots = new scala.collection.mutable.ListBuffer[Long]()
+      startTime = System.currentTimeMillis()
+      intermediate = startTime
+      randomOrderList.zipWithIndex.foreach { case (tx, idx) =>
+        nodeViewHolder.txModify(tx.asInstanceOf[SidechainTypes#SCAT])
+        if ((idx + 1) % numOfTxsPerSnapshot == 0) {
+          val updateTime = System.currentTimeMillis()
+          listOfSnapshots += (updateTime - intermediate)
+          intermediate = updateTime
+        }
+      }
+      totalTime = System.currentTimeMillis() - startTime
+      assertEquals(numOfTxs, mempool.size)
+
+      out.write(s"\n********************* Random order test results *********************\n")
+
+      println(s"Total time $totalTime ms")
+      timePerTx = totalTime.toFloat / numOfTxs
+      println(s"Average time per transaction ${timePerTx} ms")
+      println(
+        s"Average time per transactions in Snapshots ${listOfSnapshots.map(res => res.toFloat / numOfTxsPerSnapshot).mkString(",")} "
+      )
+      out.write(s"Duration of the test:                      $totalTime ms\n")
+      out.write(s"Average time per transaction:             ${timePerTx} ms\n")
+      out.write(s"Average time per transaction in snapshots:\n")
+      listOfSnapshots.map(res => res.toFloat / numOfTxsPerSnapshot).zipWithIndex.foreach { case (res, idx) =>
+        out.write(s"Snapshot $idx: $res ms\n")
+      }
+      out.write(s"Number of transactions per snapshot: $numOfTxsPerSnapshot\n")
+
+
+    } finally {
+      out.write("\n\n")
+      out.close()
+    }
+  }
+
+  /*
+    Same as txModifyTest but it takes into account the nonce gap. So the maximum number of txs for each account can be
+    only 16. The other test is temporarily kept just for comparing the performances with the previous implementations
+    of the mempool.
+   */
+  @Test
+  //@Ignore
```

Comment:
don't forget to move all the performance tests to `Ignore` again.

Reply:
Done